packages/react-styles/scripts/generateClassMaps.js

```diff
@@ -0,0 +1,74 @@
+const path = require('path');
+const fs = require('fs-extra');
+const glob = require('glob');
+const camelcase = require('camel-case');
+
+/**
+ * @param {string} cssString - CSS string
+ */
+function getCSSClasses(cssString) {
+  return cssString.match(/(\.)(?!\d)([^\s.,{[>+~#:)]*)(?![^{]*})/g);
```

Comment:
Is there any way we can use some CSS parser library to pull out these class names, instead of our own regex? Maybe that would be less performant. I'm hesitant to take on maintaining stuff like this in case it's missing some corner case or could be broken by future changes to the CSS spec (however unlikely).

Maybe we could use something like one of these, and then walk the resulting AST:
* https://www.npmjs.com/package/css-selector-parser (https://stackoverflow.com/a/38422908)
* https://www.npmjs.com/package/css
* https://www.npmjs.com/package/string-extract-class-names

Reply:
I just copied this over from our existing build script. I think it would be fantastic to leverage https://www.npmjs.com/package/css which we already use.

> Also, can we convert these scripts to TS for the sake of maintainability, or does that complicate the build too much?

We could, but this would also require us using `ts-node` instead of `node`. We have maybe 20 or so such script files in the repo that it'd be good to convert all of them.